junit-platform-launcher/src/main/java/org/junit/platform/launcher/tagexpression/Operator.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.launcher.tagexpression;
+
+import static org.junit.platform.launcher.tagexpression.Associativity.Left;
+import static org.junit.platform.launcher.tagexpression.ParseStatus.missingOperatorBetween;
+import static org.junit.platform.launcher.tagexpression.ParseStatus.missingRhsOperand;
+import static org.junit.platform.launcher.tagexpression.ParseStatus.problemParsing;
+import static org.junit.platform.launcher.tagexpression.ParseStatus.success;
+
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+class Operator {
+
+	static Operator nullaryOperator(String representation, int precedence) {
+		return new Operator(representation, precedence, 0, null, (expressions, position) -> success());
+	}
+
+	static Operator unaryOperator(String representation, int precedence, Associativity associativity,
+			Function<Expression, Expression> unaryExpression) {
+		return new Operator(representation, precedence, 1, associativity, (expressions, position) -> {
+			Position<Expression> rhs = expressions.pop();
+			if (position < rhs.position) {
+				Expression not = unaryExpression.apply(rhs.element);
```

Comment:
Since this is more general, it should *not* be called `not`... 😉

Reply:
inlined not to keep make it more similar to binaryOperator